### PR TITLE
Update min PyTorch version to 1.9, resolve deprecation warnings

### PR DESCRIPTION
**Pull request by gpleiss**
There are a number of new deprecation warnings due to PyTorch 1.10 (plus some other warnings that we've previously missed). This PR addresses all warnings that pop up in the test suite.

This should hopefully be an easy merge.

### CODE REVIEW

**Comment by Balandat** [comment]
It seems a bit aggressive to jump from 1.8.1 min req to 1.10 skipping 1.9, and just one month after 1.10 was released. This could mean a lot of folks that are using this in a prod setting and who may be able to upgrade gpytorch but not torch would not be able to use this for a while. Are there any warnings that were issued starting with 1.9? If so coudl we split those out and make a release that requires 1.9, and then merge the rest of this?

**Comment by gpleiss** [comment]
Okay yeah that's fair. There was only one minor 1.10-specific warning. I'll split that out

File: gpytorch/lazy/block_diag_lazy_tensor.py [hunk 1]
```diff
@@ -52,8 +52,8 @@ def _cholesky_solve(self, rhs, upper: bool = False):
 
     def _get_indices(self, row_index, col_index, *batch_indices):
         # Figure out what block the row/column indices belong to
-        row_index_block = row_index // self.base_lazy_tensor.size(-2)
-        col_index_block = col_index // self.base_lazy_tensor.size(-1)
+        row_index_block = torch.div(row_index, self.base_lazy_tensor.size(-2), rounding_mode="floor")
+        col_index_block = torch.div(col_index, self.base_lazy_tensor.size(-1), rounding_mode="floor")
```
Comment (Balandat) [inline gpytorch/lazy/block_diag_lazy_tensor.py:56]:
```suggestion
        row_size, col_size = self.base_lazy_tensor.shape[-2:]
        row_index_block = torch.div(row_index, row_size, rounding_mode="floor")
        col_index_block = torch.div(col_index, col_size, rounding_mode="floor")
```

File: gpytorch/lazy/block_interleaved_lazy_tensor.py [hunk 2]
```diff
@@ -53,8 +53,8 @@ def _get_indices(self, row_index, col_index, *batch_indices):
         col_index_block = col_index.fmod(self.base_lazy_tensor.size(-3))
 
         # Find the row/col index within each block
-        row_index = row_index // self.base_lazy_tensor.size(-3)
-        col_index = col_index // self.base_lazy_tensor.size(-3)
+        row_index = torch.div(row_index, self.base_lazy_tensor.size(-3), rounding_mode="floor")
+        col_index = torch.div(col_index, self.base_lazy_tensor.size(-3), rounding_mode="floor")
```
Comment (Balandat) [inline gpytorch/lazy/block_interleaved_lazy_tensor.py:57]:
```suggestion
        block_dim = self.base_lazy_tensor.size(-3)
        row_index = torch.div(row_index, block_dim, rounding_mode="floor")
        col_index = torch.div(col_index, block_dim, rounding_mode="floor")
```

**Review by Balandat — APPROVED** [review]
couple of nits, o/w lgtm!

**Comment by Balandat** [comment]
oops so much for the suggestions, auto-merge

**Pull request merged**